provider/mem/mem.go

```diff
@@ -100,35 +101,57 @@ func NewAlerts(ctx context.Context, m types.Marker, intervalGC time.Duration, al
 		logger:    log.With(l, "component", "provider"),
 		callback:  alertCallback,
 	}
-	a.alerts.SetGCCallback(func(alerts []*types.Alert) {
-		for _, alert := range alerts {
+	if r != nil {
+		a.registerMetrics(r)
+	}
+
+	go a.gcLoop(ctx, intervalGC)
+
+	return a, nil
+}
+
+func (a *Alerts) gcLoop(ctx context.Context, interval time.Duration) {
+	t := time.NewTicker(interval)
+	defer t.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			a.gc()
+		}
+	}
+}
+
+func (a *Alerts) gc() {
```

Comment:
nit: just a style question, should we put this gc function not belong to Alerts type as this implementation is specific to mem.go class

Reply:
originally, mem.go used a separate goroutine for this task. However, this is not appropriate in this context because we need the lock protection from Alerts. 

I agree that we can create a non-async GC function in mem.go.